external-crates/move/documentation/book/src/control-flow/pattern-matching.md

```diff
@@ -0,0 +1,618 @@
+        MyEnum::Variant(1, true) => 1,
+        MyEnum::OtherVariant(_, 3) => 2,
+        MyEnum::Variant(..) => 3,
+        MyEnum::OtherVariant(..) => 4,
+}
+f(MyEnum::Variant(1, true)); // returns 1
+f(MyEnum::Variant(2, true)); // returns 3
+f(MyEnum::OtherVariant(false, 3)); // returns 2
+f(MyEnum::OtherVariant(true, 3)); // returns 2
+f(MyEnum::OtherVariant(true, 2)); // returns 4
+```
+
+This is saying that "if `x` is `MyEnum::Variant` with the fields `1` and `true`, then return `1`, if
+it is `MyEnum::OtherVariant` with any value for the first field, and `3` for the second, then return
+`2`, if it is `MyEnum::Variant` with any fields, then return `3`, and if it is
+`MyEnum::OtherVariant` with any fields, then return `4`".
+
+You can also nest patterns, so if you wanted to match either 1, 2, or 10, instead of just matching 1
+in the `MyEnum::Variant` above, you could do so with an or-pattern:
+
+```move
+fun f(x: MyEnum) {
+    match (x) {
+        MyEnum::Variant(1 | 2 | 10, true) => 1,
+        MyEnum::OtherVariant(_, 3) => 2,
+        MyEnum::Variant(..) => 3,
+        MyEnum::OtherVariant(..) => 4,
+}
+f(MyEnum::Variant(1, true)); // returns 1
+f(MyEnum::Variant(2, true)); // returns 1
+f(MyEnum::Variant(10, true)); // returns 1
+f(MyEnum::Variant(10, false)); // returns 3
+```
+
+### Ability Constraints
+
+Additionally, match bindings are subject to the same ability restrictions as other aspects of Move.
+In particular, the compiler will signal an error if you try to match a value (i.e., not-reference)
+without `drop` using a wildcard, as the wildcard expects to drop the value. Similarly, if you bind a
+non-`drop` value using a binder, it must be used in the right-hand side of the match arm. In
+addition, if you fully-destruct that value, you have unpacked it, matching the semantics of
+[non-`drop` struct unpacking](link). See [ref section] for more details about the `drop` capability.
+
+```move
+public struct NonDrop(u64)
+
+fun drop_nondrop(x: NonDrop) {
+    match (x) {
+        NonDrop(1) => 1,
+        _ => 2
+        // ERROR: cannot wildcard match on a non-droppable value
+    }
+}
+
+fun destructure_nondrop(x: NonDrop) {
+    match (x) {
+        NonDrop(1) => 1,
+        NonDrop(_) => 2
+        // OK!
+    }
+}
+
+fun use_nondrop(x: NonDrop): NonDrop {
+    match (x) {
+        NonDrop(1) => NonDrop(8),
+        x => x
+    }
+}
+```
+
+## Exhaustiveness
+
+The `match` expression in Move must be _exhaustive_: every possible value of the type being matched
+must be covered by one of the patterns in one of the match's arms. If the series of match arms is
+not exhaustive, the compiler will raise an error. Note that any arm with a guard expression does not
+contribute to match exhaustion, as it may fail to match at runtime.
+
+As an example, if we were to match on a `u8` then in order for the match to be exhaustive we would
+need to match on _every_ number from 0 to 255 inclusive, or a wildcard or variable pattern would
```

Comment:
```suggestion
it matches on _every_ number from 0 to 255 inclusive, unless there is a wildcard or variable pattern present.
```
I might have accidentally changed the meaning here...

Reply:
Reads correctly to me still!